debugtools/DDR_VM/src/com/ibm/j9ddr/tools/ddrinteractive/BufferedMemory.java

```
@@ -0,0 +1,225 @@
+ *   J9ROMMethodPointer romMethod = pointer.romMethods();
+ *   long dumpFlags = (ByteOrder.nativeOrder() == ByteOrder.BIG_ENDIAN) ? 1 : 0;
+ *   J9BCUtil.j9bcutil_dumpRomMethod(System.out, romMethod, pointer, dumpFlags, J9BCUtil.BCUtil_DumpAnnotations);
+ * }
+ * </code>
+ *
+ * Behaviour of the model is not defined
+ * for the case where the underlying source's
+ * ByteBuffer is modified (address or capacity)
+ * during use.
+ *
+ * @author knewbury01
+ *
+ */
+
+public class BufferedMemory extends AbstractMemory implements IProcess, IAddressSpace
+{
+	public BufferedMemory(ByteOrder byteOrder) {
+		super(byteOrder);
+	}
+
+	@Override
+	public Platform getPlatform() {
+		String platform = System.getProperty("os.name").toLowerCase();
+		if (platform.contains("aix")) {
+			return Platform.AIX;
+		}
+		else if (platform.contains("windows")) {
+			return Platform.WINDOWS;
+		}
+		else if	(platform.contains("z/os")) {
+			return Platform.ZOS;
+		}
+		else if	(platform.contains("linux")) {
+			return Platform.LINUX;
+		}
+		else if (platform.contains("mac")) {
+			return Platform.OSX;
+        }
+		else {
+			//do not expect to reach here
+			return null;
+		}
+	}
+
+	/**
+	 *
+	 * @return Address space this process uses.
+	 */
+	@Override
+	public IAddressSpace getAddressSpace() {
+		return this;
+	}
+
+	@Override
+	public long getPointerAt(long address) throws MemoryFault {
+		if (bytesPerPointer() == 8) {
+			return getLongAt(address);
+		} else {
+			return (0xFFFFFFFFL & getIntAt(address));
+		}
+	}
+
+	/**
+	 *
+	 * @return Number of bytes in a pointer
+	 */
+	@Override
+	public int bytesPerPointer() {
+		return Integer.parseInt(System.getProperty("sun.arch.data.model"));
+	}
+
+	/**
+	 * @return Process command line or null if the data is unavailable
+	 * @throws CorruptDataException
+	 */
+	@Override
+	public String getCommandLine() throws DataUnavailableException, CorruptDataException {
+		return null;
```

This should throw a `DataUnavailableException` rather than return null.